src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/xunitrunner-perf/csvjsonconvertor.py

```diff
@@ -0,0 +1,195 @@
+    job.user = user
+
+    buildInfo = serialobj.BuildInfo()
+    buildInfo.buildInfoName = opts['--buildInfoName']
+    buildInfo.buildNumber = opts['--buildNumber']
+    buildInfo.branch = opts['--branch']
+    job.buildInfo = buildInfo
+
+    jobType = serialobj.JobType()
+    jobType.jobTypeName = opts['--jobTypeName']
+    job.jobType = jobType
+
+    jobGroup = serialobj.JobGroup()
+    jobGroup.jobGroupName = opts['--jobGroupName']
+    job.jobGroup = jobGroup
+
+    job.jobDescription = opts['--jobDescription']
+    job.jobName = opts['--jobName']
+
+    root = serialobj.Root()
+    root.job = job
+    jsonOutput = serialobj.JsonOutput()
+    jsonOutput.roots.append(root)
+    with open(opts['--jsonFile'], 'w+') as opfile:
+        opfile.write(jsonOutput.to_JSON())
+        opfile.flush()
+        opfile.close()
+
+    log.info('Conversion of csv to json successful')
+
+def checkRequiredArgs(optdict):
+    if not '--csvFile' in optdict or not '--jobName' in optdict or not '--configName' in optdict or not '--userAlias' in optdict or not '--branch' in optdict or not '--osInfoName' in optdict or not '--jsonFile' in optdict:
+        raise getopt.GetoptError
+
+def validateArgs(args):
+    # validate required params
+    opts, args = getopt.getopt(args, '', ['csvFile=', 'jsonFile=', 'jobName=', 'jobDescription=', 'configName=', 'jobGroupName=', 'jobTypeName=', 'username=', 'userAlias=', 'branch=', 'buildInfoName=', 'buildNumber=', 'machinepoolName=', 'machinepoolDescription=', 'architectureName=', 'manufacturerName=' , 'microarchName=', 'numberOfCores=', 'numberOfLogicalProcessors=', 'totalPhysicalMemory=', 'osInfoName=', 'osVersion=', 'machineName=', 'machineDescription='])
+    optdict = dict(opts)
+    checkRequiredArgs(optdict)
+    return optdict
+
+def main():
+    """
+        Usage::
+            csv_to_json.py
+                --csvFile csvFile to convert to json
+                --jsonFile output json file location
+                --jobName sample job
+                --jobDescription sample job description
+                --configName sample config
+                --jobGroupName sample job group
+                --jobTypeName Private
+                --username sample user
+                --userAlias sampleuser
+                --branch ProjectK
+                --buildInfoName sample build
+                --buildNumber 1234
+                --machinepoolName HP Z210 Workstation
+                --machinepoolDescription Intel64 Family 6 Model 42 Stepping 7
+                --architectureName AMD64
+                --manufacturerName Intel
+                --microarchName SSE2
+                --numberOfCores 4
+                --numberOfLogicalProcessors 8
+                --totalPhysicalMemory 16342
+                --osInfoName Microsoft Windows 8.1 Pro
+                --osVersion 6.3.9600
+                --machineName JORIVE-PC
+                --machineDescription Intel(R) Core(TM) i7-2600 CPU @ 3.40GHz
```

Comment:
I assume these arguments almost certainly need to be quoted, so the example should probably say that.

It's also somewhat unusual for "usage" to have values for the various arguments, usually they are left as placeholders describing what they are.